### PR TITLE
drivers: wifi: eswifi: Offload sockets accessing invalid net_context

### LINKED ISSUE #52317: drivers: wifi: eswifi: Offload sockets accessing invalid net_context, resulting in errors in FLASH_SR and blocking flash use
**Describe the bug**

It appears that eswifi driver may have an implementation issue when using offloaded sockets. I found the issue by trying to write to flash memory after opening a socket, but my writes returned with -EIO errors. I found the FLASH_SR status register reported errors with value 0xe0, and traced it to a call in the eswifi driver. 

The call that triggers this is [net_context_set_state](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/wifi/eswifi/eswifi_socket.c#L224), which silently fails to update the state flag on the context object. The context object is located at a very low address, **0x50ce**, which is the value of [ESWIFI_INIT_CONTEXT](https://github.com/zephyrproject-rtos/zephyr/blob/c27a6af7128a433fca825c43a0efe2f9405892b5/drivers/wifi/eswifi/eswifi_socket_offload.c#L33), or the which I believe is a dummy socket.

In Discord networking channel, @rlubos suggested that `struct net_context` is not used in the offloaded socket scenario, but apparently the offload socket implementation accesses the context anyway even though the context just the dummy socket at 0x50ce.

Please also mention any information which could help others to understand the problem you're facing:
 - What target platform are you using? STM32 - B-L4S5I-IOT01A
 - What have you tried to diagnose or workaround this issue? I am simply clearing the FLASH_SR register after opening a socket to unblock my objectives writing to flash memory after fetching data.
 - Is this a regression? If yes, have you been able to "git bisect" it to a
   specific commit? Possibly: https://github.com/zephyrproject-rtos/zephyr/commit/cb1d588ea737fdb7be7ee0c3eb16f8d13c7c03a7

**To Reproduce**
1. Enable the WiFi Shell 
2. Use the http get sample. Edit it so that it only fetches after wifi has been connected (e.g. with delay). Also add an edit that after the socket is connected, the value of the following register is printed: `volatile uint32_t* flash_sr = 0x40022010;`
3. Flash the board, and using the shell connect to wifi and run the http demo

Alternatively, you could probably trigger the same issue using the shell alone. Connect to wifi, use net shell to open a socket, and then gdb to inspect the FLASH_SR register.

**Expected behavior**
A) The FLASH_SR register should be 0, but in the error case, is 0xe0
B) The network context->flags member should be updated with the CONNECTED state, but in the error case it is the same as before (you may need to add more debugging statements or use gdb to view this).

**Impact**
I have a workaround, clear the flash sr register and keep chugging. But the network context state flag is incorrect, and I do not fully know what the implications of that are.

**Logs and console output**

After opening a socket (specifically after setting network state flag)

```
(gdb) p/x *(uint32_t*) 0x40022010
 $15 = 0xe0
```

**Environment (please complete the following information):**
 - OS: Ubuntu
 - Toolchain Zephyr SDK 0.15.1
 - V3.2.0

### PR DESCRIPTION
**Pull request by bdunlay**
This change fixes a regression from commit 1cbc0acd7e5f5d8c56bc66a475a76358db12a1b6.

Socket offload uses a dummy socket context, and setting the socket state in this dummy context is invalid.

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/52317

Signed-off-by: Brian Dunlay <brian@nubix.io>

### CODE REVIEW

**Comment by henrikbrixandersen** [comment]
@bdunlay You need to fix the commit log to have shorter a line.

**Comment by bdunlay** [comment]
I am actually thinking that perhaps this call should be conditionally defined based on `CONFIG_NET_SOCKETS_OFFLOAD` because the call to `__eswifi_off_start_client` can come from wifi offload OR socket offload, meaning that the socket context can in some circumstances be valid.

For example on line 224

```
#ifdef CONFIG_NET_SOCKETS_OFFLOAD
net_context_set_state(socket->context, NET_CONTEXT_CONNECTED);
#endif
```

**Review by tbursztyka — APPROVED** [review]

**Review by nandojve — APPROVED** [review]

**Comment by bdunlay** [comment]
Requesting feedback from an eswifi maintainer before merge. Please see my previous comment.

**Pull request merged**
